lib/ansible/plugins/lookup/fileglob.py

```diff
@@ -26,12 +26,10 @@ class LookupModule(LookupBase):
 
     def run(self, terms, variables=None, **kwargs):
 
-        basedir = self.get_basedir(variables)
-
         ret = []
         for term in terms:
             term_file = os.path.basename(term)
-            dwimmed_path = self._loader.path_dwim_relative(basedir, 'files', os.path.dirname(term))
+            dwimmed_path = self.find_needle(variables, 'files', os.path.dirname(term))
```

Comment:
no, this is very wrong as is, we need to select which subdir depending on task action, otherwise include_vars will try to read an template with the same name